hazelcast/src/main/java/com/hazelcast/wan/merkletree/ArrayMerkleTree.java

```diff
@@ -94,16 +94,16 @@
     private volatile long footprint;
 
     @SuppressWarnings("unchecked")
-    public ArrayMerkleTree(int depth) {
+    public ArrayMerkleTree(int depth, LeafSetFactory leafSetFactory) {
         super(depth);
 
         this.leafLevel = depth - 1;
 
         final int leaves = MerkleTreeUtil.getNodesOnLevel(leafLevel);
 
-        leafKeys = new OAHashSet[leaves];
+        leafKeys = new HashAcceptorSet[leaves];
         for (int i = 0; i < leaves; i++) {
-            leafKeys[i] = new OAHashSet<Object>(1);
+            leafKeys[i] = leafSetFactory.createLeafSet(5);
```

Comment:
This change is important (initial capacity 1 -> 5). Higher values mean bigger heap consumption in exchange for less frequent rehashing. If dozens of records are put into a map, big enough initial capacity takes a significant pressure off from the GC. We should consider making this (and the set's load factor) configurable.